packages/cli/src/commands/doctor/healthchecks/iosDeploy.ts

```diff
@@ -1,58 +1,116 @@
+const installationWithNpm = 'npm install ios-deploy --global';
+
+const identifyInstallationCommand = () => {
   if (packageManager === PACKAGE_MANAGERS.YARN) {
-    return 'yarn global add ios-deploy';
+    return installationWithYarn;
   }
 
   if (packageManager === PACKAGE_MANAGERS.NPM) {
-    return 'npm install ios-deploy --global';
+    return installationWithNpm;
   }
 
   return undefined;
 };
 
+const installLibrary = async ({
+  installationCommand,
+  packageManagerToUse,
+  loader,
+}: {
+  installationCommand: string;
+  packageManagerToUse: 'yarn' | 'npm';
+  loader: Ora;
+}) => {
+  try {
+    loader.start(`${label} (installing with ${packageManagerToUse})`);
+
+    const installationCommandArgs = installationCommand.split(' ');
+
+    await execa(installationCommandArgs[0], installationCommandArgs.splice(1));
+
+    loader.succeed(`${label} (installed with ${packageManagerToUse})`);
+  } catch (_error) {
+    loader.fail();
+
+    logManualInstallation({
+      healthcheck: 'ios-deploy',
+      command: installationCommand,
+    });
+  }
+};
+
 export default {
-  label: 'ios-deploy',
+  label,
   isRequired: false,
   getDiagnostics: async () => ({
     needsToBeFixed: await checkSoftwareInstalled('ios-deploy'),
   }),
   runAutomaticFix: async ({loader}) => {
-    const installationCommand = getInstallationCommand();
+    loader.stop();
+
+    const installationCommand = identifyInstallationCommand();
 
     // This means that we couldn't "guess" the package manager
     if (installationCommand === undefined) {
-      loader.fail();
+      const promptQuestion = `ios-deploy needs to be installed either by ${chalk.bold(
+        'yarn',
+      )} ${chalk.reset('or')} ${chalk.bold(
+        'npm',
+      )} ${chalk.reset()}, which one do you want to use?`;
+      const installWithYarn = 'yarn';
+      const installWithNpm = 'npm';
+      const skipInstallation = 'Skip installation';
 
-      // Then we just print out the URL that the user can head to download the library
-      logManualInstallation({
-        healthcheck: 'ios-deploy',
-        url: 'https://github.com/ios-control/ios-deploy#readme',
-      });
-      return;
-    }
+      const {whichPackageManagerToUse} = await inquirer.prompt([
+        {
+          type: 'list',
+          name: 'whichPackageManagerToUse',
+          message: promptQuestion,
+          choices: [installWithYarn, installWithNpm, skipInstallation],
+        },
+      ]);
+
+      removeMessage(`? ${promptQuestion} ${whichPackageManagerToUse}`);
 
-    try {
-      const installationCommandArgs = installationCommand.split(' ');
+      if (whichPackageManagerToUse === skipInstallation) {
+        loader.fail();
 
-      await execa(
-        installationCommandArgs[0],
-        installationCommandArgs.splice(1),
-      );
+        // Then we just print out the URL that the user can head to download the library
+        return logManualInstallation({
```

Comment:
if `logManualInstallation` doesn't return, please revert this change, I prefer being explicit with returns